pkg/cincinnati/cincinnati.go

```diff
@@ -146,17 +156,98 @@ func (c Client) GetUpdates(ctx context.Context, uri *url.URL, arch string, chann
 		}
 	}
 
-	var updates []Update
+	var updates []configv1.Release
 	for _, i := range nextIdxs {
-		updates = append(updates, Update(graph.Nodes[i]))
+		update, err := convertRetrievedUpdateToRelease(graph.Nodes[i])
+		if err != nil {
+			return current, nil, nil, &Error{
+				Reason:  "ResponseInvalid",
+				Message: fmt.Sprintf("invalid recommended update node: %s", err),
+			}
+		}
+		updates = append(updates, update)
+	}
+
+	var conditionalUpdates []configv1.ConditionalUpdate
+	for _, conditionalEdges := range graph.ConditionalEdges {
+		for _, edge := range conditionalEdges.Edges {
+			if version.String() == edge.From {
+				var target *node
+				for i, node := range graph.Nodes {
+					if node.Version.String() == edge.To {
+						target = &graph.Nodes[i]
+						break
+					}
+				}
+				if target == nil {
+					return current, updates, nil, &Error{
+						Reason:  "ResponseInvalid",
+						Message: fmt.Sprintf("no node for conditional update %s", edge.To),
```

Comment:
In the case target node is null, it seems `edge.To` would be null as well. If that's true, seems that's no need to log it in the message.

Reply:
`edge.To` might have a version string that just happens to not be listed in `nodes`.  Not something that should exist in well-formed Cincinnati graph JSON, but still worth logging.

Having `edge.To` be empty would also be a problem, and we could also complain about that without even looking through `nodes`.  As this code stands, I guess we could have an empty `edge.To` and an entry in `nodes` with an empty-string `version` and maybe not get a `ResponseInvalid` complaint?  Seems fair to open a bug if we aren't setting `ResponseInvalid` if `nodes[].version == ""` or unset today.